packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -305,6 +310,33 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
     }
   }, [apiRef, clearFilteredRows, logger, rowsProp]);
 
+  const prevFilterModelRef = React.useRef();
+
+  React.useEffect(() => {
+    return apiRef.current.subscribeEvent(GRID_FILTER_MODEL_CHANGE, (params) => {
```

Comment:
it will look cleaner if you move that function in its own definition, like `handleFilterModelChange` or smth.